optaplanner-core/src/main/java/org/optaplanner/core/impl/solver/DefaultSolverFactory.java

```diff
@@ -97,8 +97,8 @@ public DefaultSolverFactory(SolverConfig solverConfig) {
         Termination<Solution_> termination = TerminationFactory.<Solution_> create(terminationConfig_)
                 .buildTermination(configPolicy, basicPlumbingTermination);
         List<Phase<Solution_>> phaseList = buildPhaseList(configPolicy, bestSolutionRecaller, termination);
-        return new DefaultSolver<>(environmentMode_, randomFactory,
-                bestSolutionRecaller, basicPlumbingTermination, termination, phaseList, solverScope);
+        return new DefaultSolver<>(environmentMode_, randomFactory, bestSolutionRecaller, basicPlumbingTermination,
+                termination, phaseList, solverScope, moveThreadCount_ == null ? 1 : moveThreadCount_);
```

Comment:
If `moveThreadCount` is `null`, is it really a single move thread or rather 0, as the moves are evaluated directly on the solver thread?

Reply:
Good question. Rather a philosophical one, I'd say. There will be 1 thread.
Does it make sense to say there would be 0 move threads?

Reply:
None of these options is accurate. How about:
`move thread count (solver thread)` in case the `moveThreadCount == 0` ?

Reply:
when moveThreadCount is null, we use NONE which is != 1.

With NONE, the moves are evaluated in the solver thread. with 1, the moves are evaluated in 1 move thread on which the solver thread is waiting.
move thread count 1 is mainly usefull for debugging purposes, not for in production. NONE is always better.

Reply:
```suggestion
                termination, phaseList, solverScope, moveThreadCount_ == null ? SolverConfig.MOVE_THREAD_COUNT_NONE : Integer.toString(moveThreadCount_));
```